runtime/onert/core/src/exporter/train/CheckpointExporter.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "exporter/train/CheckpointExporter.h"
+
+#include "exec/Execution.h"
+#include "ir/train/TrainingInfo.h"
+#include "ir/train/Checkpoint.h"
+
+#include <fstream>
+#include <iostream>
+#include <numeric>
+
+namespace
+{
+
+using namespace onert;
+using namespace ir;
+using namespace train;
+using namespace checkpoint;
+using namespace exec;
+
+class CheckpointExporter
```

Comment:
(Out of curiosity)
Can I know the reason why this class hides on cc file?

Reply:
This is because this class can only be accessed through `exportCheckpoint` api. If this class is accessed outside of this file, I will move this class under the onert::ir::train namespace. :)